enhancements/network/ovn-k-localnet-api.md

```diff
@@ -0,0 +1,514 @@
+---
+title: user-defined-network-segmentation
+authors:
+  - "@ormergi"
+reviewers:
+  - "@maiqueb"
+  - TBD
+approvers:
+  - TBD
+api-approvers:
+creation-date: 2025-02-04
+last-updated: 2025-02-04
+tracking-link:
+  - https://issues.redhat.com/browse/SDN-5519
+---
+
+# OVN-Kubernetes Localnet API
+
+## Summary
+
+Extend OVN-Kubernetes ClusterUserDefinedNetwork (CUDN) API to support localnet topology.
+
+## Motivation
+
+As of today one can create a user-defined network over localnet topology using NetworkAttachmentDefinition (NAD).
+Using NAD for localnet has some pitfalls due to the fact it is not managed and not validated on creation.
+Misconfigurations are detected too late causing bad UX and frustration for users.
+
+Configuring localnet topology networks require changes to cluster nodes network stack and involve some risk and 
+knowledge that require cluster-admin intervention.
+Such as configuring the OVS switch to which the localnet network connects to. 
+
+Managing localnet topology networks using a well-formed API using the CUDN CRD could improve UX as it is managed by a 
+controller, perform validations and reflect the state via status.
+
+### User Stories
+
+#### Definition of personas:
+Admin - is the cluster admin.
+User - non cluster-admin user, project manager.
+Workloads - pod or VMs (KubeVirt).
+
+- As an admin I want to create a user-defined network over localnet topology using CUDN CRD. 
+  - In case the network configuration is bad I want to get an informative message saying what went wrong.
+- As an admin I want to enable users to connect workloads in project/namespaces they have permission to, to the localnet network I created for them.
+- As a user I want to be able to connect my workloads (pod/VMs) to the localnet the admin created in my namespace.
+- As a user I want my workloads to be able to communicate with each other over the localnet network.
+- As a user I want my connected VMs to the localnet network to be able to migrate from one node to another, having its localnet network interface IP address unchanged.
+
+### Goals
+
+- Enable creating user-defined-networks over localnet topology using OVN-K CUDN CRD.
+- Streamline localnet UX: detect misconfigurations early, communicate issues via informative status conditions and or events.
+
+### Non-Goals
+
+- This proposal does not cover changes for network configurations on day2 (mutable network configuration).
+
+## Proposal
+
+### Summary
```

Comment:
another summary ?

Reply:
I tried to express the proposal in few sentences to enable reviewers realize whats the proposed solution idea.